### PR TITLE
Bump minimum node version to 16.11

### PR DESCRIPTION
**Pull request by samchungy**
Shall we bump this now considering we're releasing a bunch of stuff anyway to avoid doing another update very soon?

### CODE REVIEW

File: .changeset/empty-donuts-arrive.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+deps: Require Node.js 16+
+
+Node.js 14 will reach end of life by April 2023
+
+Consider upgrading the Node.js version for your project across:
+
+.nvmrc
+package.json#/engines/node
+CI/CD configuration (.buildkite/pipeline.yml, Dockerfile, etc.)
```
Comment (72636c) [inline .changeset/empty-donuts-arrive.md:13]: Just stealing formatting from https://github.com/seek-oss/skuba/releases/tag/v4.0.0:

```suggestion
Node.js 14 will reach end of life by April 2023.

Consider upgrading the Node.js version for your project across:

- `.nvmrc`
- `package.json#/engines/node`
- CI/CD configuration (`.buildkite/pipeline.yml`, `Dockerfile`, etc.)
```

**Review by 72636c — COMMENTED** [review]
Good idea. Should we require a recent minor version?

- [16.11+ should give us ES2022](https://node.green/#ES2022)
- [16.13+ is LTS](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V16.md#16.19.1)

**Review by 72636c — APPROVED** [review]
Is it worth bumping the npm-package templates to target ES2022 as well? https://github.com/search?q=repo%3Aseek-oss%2Fskuba+ES2020+language%3A%22JSON+with+Comments%22&type=code&l=JSON+with+Comments

**Comment by samchungy** [comment]
> Is it worth bumping the npm-package templates to target ES2022 as well? https://github.com/search?q=repo%3Aseek-oss%2Fskuba+ES2020+language%3A%22JSON+with+Comments%22&type=code&l=JSON+with+Comments

@72636c if we do that, should we be setting explicit engine versions in the package json?

eg. 
```ts
  "engines": {
    "node": ">=16.11"
  },
```

File: template/private-npm-package/_package.json
Comment (samchungy) [inline template/private-npm-package/_package.json:1]: Sorted the package json files

File: template/private-npm-package/_package.json [hunk 2]
```diff
@@ -35,7 +32,13 @@
...
-  "types": "./lib-types/index.d.ts",
-  "version": "0.0.0-semantically-released"
+  "dependencies": {},
+  "devDependencies": {
+    "@types/node": "^16.18.3",
+    "commitizen": "^4.2.4",
+    "skuba": "*"
+  },
+  "engines": {
+    "node": ">=16.11"
```
Comment (samchungy) [inline template/private-npm-package/_package.json:42]: only change is this

**Review by 72636c — APPROVED** [review]
🙇

**Pull request merged**